### PR TITLE
ListOffsetsRequest should only be sent to the leader replica (CI)

### PR DESCRIPTION
**Pull request by emasab**
Continuation of https://github.com/confluentinc/librdkafka/pull/4616 from an internal branch due to CI issues

### CODE REVIEW

**Comment by mjd95** [comment]
@emasab do you think this PR could be merged soon? I saw there were some CI issues on previous iterations but this one looks green

File: tests/0104-fetch_from_follower_mock.c [hunk 1]
```diff
@@ -111,6 +110,11 @@ static void do_test_offset_reset(const char *auto_offset_reset) {
         else
                 test_consumer_poll(auto_offset_reset, c, 0, 1, 0, msgcnt, NULL);
 
+        /* send another batch of messages to ensure the consumer isn't stuck */
```
Comment (milindl) [inline tests/0104-fetch_from_follower_mock.c:113]: I think it's a good idea to expand on why exactly we add this and why the consumer was getting stuck, alternatively, we can provide the issue # . Without that, it'll be a bit difficult for future readers.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,7 +2,10 @@
 
 librdkafka v2.5.0 is a feature release.
 
-* Fix segfault when using long client id because of erased segment when using flexver. (#4689)
+ * Update bundled lz4 (#4726)
+ * Fix segfault when using long client id because of erased segment when using flexver. (#4689)
+ * Fix for a loop of ListOffset requests, happening in a Fetch From Follower
```
Comment (milindl) [inline CHANGELOG.md:7]: Fix changelog for this to be in 2.5.1

**Review by milindl — APPROVED** [review]
Thanks @kphelps and @emasab . Looks good to me bar a few minor documentation related things.

**Review by milindl — APPROVED** [review]

**Pull request merged**